test/deploy_changes/common/common_helpers.go

```diff
@@ -0,0 +1,57 @@
+package common
+
+import (
+	"bytes"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/loft-sh/vcluster/test/framework"
+	"github.com/onsi/gomega"
+)
+
+const (
+	pollingInterval     = time.Second * 2
+	pollingDurationLong = time.Minute * 2
+	filePath            = "commonValues.yaml"
+	chartPath           = "../../../chart"
+)
+
+func ReplaceYAMLPlaceholders() {
+	replaceCmd := exec.Command("sed", "-i", "s|REPLACE_REPOSITORY_NAME|"+os.Getenv("REPOSITORY_NAME")+"|g", filePath)
+	err := replaceCmd.Run()
+	framework.ExpectNoError(err, "Failure to edit file")
+
+	replaceCmd = exec.Command("sed", "-i", "s|REPLACE_TAG_NAME|"+os.Getenv("TAG_NAME")+"|g", filePath)
+	err = replaceCmd.Run()
+	framework.ExpectNoError(err, "Failure to edit file")
+}
+
+func DeployChangesToVClusterUsingCLI(f *framework.Framework) {
+	gomega.Eventually(func() bool {
+		stdout := &bytes.Buffer{}
+		deployCmd := exec.Command("vcluster", "create", "--upgrade", f.VclusterName, "--namespace", f.VclusterNamespace, "--local-chart-dir", chartPath, "-f", filePath)
+		deployCmd.Stdout = stdout
+		err := deployCmd.Run()
+		framework.ExpectNoError(err)
+		return err == nil && strings.Contains(stdout.String(), "Switched active kube context to")
```

Comment:
Would an error be normal here, if so which error? We don't want to poll on an error if its not going to change.